src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -0,0 +1,1497 @@
+                {
+                    _currentIISSettings.WindowsAuthentication = value;
+                    OnPropertyChanged(nameof(WindowsAuthenticationEnabled));
+                }
+            }
+        }
+
+        private bool _removeEnvironmentVariablesRow;
+        public bool RemoveEnvironmentVariablesRow
+        {
+            get
+            {
+                return _removeEnvironmentVariablesRow;
+            }
+            set
+            {
+                OnPropertyChanged(ref _removeEnvironmentVariablesRow, value, suppressInvalidation: true);
+            }
+        }
+
+        private int _environmentVariablesRowSelectedIndex = -1;
+        public int EnvironmentVariablesRowSelectedIndex
+        {
+            get { return _environmentVariablesRowSelectedIndex; }
+            set
+            {
+                if (_environmentVariablesRowSelectedIndex != value)
+                {
+                    _environmentVariablesRowSelectedIndex = value;
+                    if (_environmentVariablesRowSelectedIndex == -1)
+                    {
+                        //No selected item - Disable Remove button
+                        RemoveEnvironmentVariablesRow = false;
+                    }
+                    else
+                    {
+                        RemoveEnvironmentVariablesRow = (EnvironmentVariablesValid) ? true : ((EnvironmentVariables[_environmentVariablesRowSelectedIndex] as NameValuePair).HasValidationError == true);
+                    }
+
+                    OnPropertyChanged(nameof(EnvironmentVariablesRowSelectedIndex), suppressInvalidation: true);
+                }
+            }
+        }
+
+        private bool _environmentVariablesValid = true;
+        public bool EnvironmentVariablesValid
+        {
+            get
+            {
+                if (EnvironmentVariables == null) { return true; }
+                else { return _environmentVariablesValid; }
+            }
+            set
+            {
+                if (_environmentVariablesValid != value)
+                {
+                    _environmentVariablesValid = value;
+                    if (value == true && ClearEnvironmentVariablesGridError != null)
+                    {
+                        ClearEnvironmentVariablesGridError.Invoke(this, EventArgs.Empty);
+                    }
+                    OnPropertyChanged(nameof(EnvironmentVariablesValid), suppressInvalidation: true);
+                }
+            }
+        }
+
+        private ObservableList<NameValuePair> _environmentVariables;
+        public ObservableList<NameValuePair> EnvironmentVariables
+        {
+            get
+            {
+                return _environmentVariables;
+            }
+            set
+            {
+                OnPropertyChanged(ref _environmentVariables, value);
+            }
+        }
+
+        protected virtual void NotifySelectedChanged(LaunchProfile oldProfile)
```

Comment:
>NotifySelectedChanged [](start = 31, length = 21)

I always hated this implementation. It more or less invalidated the entire control ( and is still buggy). I would do something better here.